symengine/functions.cpp

```diff
@@ -2647,6 +2661,29 @@ std::size_t LeviCivita::__hash__() const
     return seed;
 }
 
+RCP<const Basic> LeviCivita::subs(const map_basic_basic &subs_dict) const
+{
+    auto it = subs_dict.find(rcp_from_this());
+    if (it != subs_dict.end())
+        return it->second;
+    vec_basic arg(arg_.size());
+    bool this_rcp = true;
+    auto it2 = arg.begin();
+    for (const auto& it: arg_) {
+        *it2 = ((it)->subs(subs_dict));
+        if (this_rcp) {
```

Comment:
@isuruf 

This https://github.com/symengine/symengine/pull/836#discussion_r55921689
gives
```
symengine/symengine/functions.cpp:2676:13: error: passing ‘const Teuchos::RCP<const 
SymEngine::Basic>’ as ‘this’ argument of ‘Teuchos::RCP<T1>& Teuchos::RCP<T>::operator=
(const Teuchos::RCP<T>&) [with T = const SymEngine::Basic]’ discards qualifiers 
[-fpermissive]
     *it = t;
```

this also gives same error
```c++
    for (auto& it: arg_) {
        auto t = ((it)->subs(subs_dict));
        if (this_rcp and t != it) {
             this_rcp = false;
        }
        it = t;
    }

```